src/EduManagementLab.Api/Controllers/CoursesController.cs

```diff
@@ -130,12 +130,12 @@ public ActionResult<Course> AddCourseMembership(Guid id, Guid userId, DateTime e
 
         }
         [HttpGet]
-        [Route("{id}/GetCourseMemberships")]
-        public ActionResult<IEnumerable<Course.Membership>> GetCourseMemberships(Guid id)
+        [Route("{courseId}/GetCourseMemberships")]
+        public ActionResult<List<Course.Membership>> GetCourseMemberships(Guid courseId, bool includeMembership, bool includeUser)
```

Comment:
You don't need to expose include in the API. You know that the memberships should be fetched and can pass true to te service.